specs/electra/beacon-chain.md

```diff
@@ -277,24 +561,353 @@ def get_attesting_indices(state: BeaconState, attestation: Attestation) -> Set[V
     return output
 ```
 
+### Beacon state mutators
+
+#### Updated  `initiate_validator_exit`
+
+```python
+def initiate_validator_exit(state: BeaconState, index: ValidatorIndex) -> None:
+    """
+    Initiate the exit of the validator with index ``index``.
+    """
+    # Return if validator already initiated exit
+    validator = state.validators[index]
+    if validator.exit_epoch != FAR_FUTURE_EPOCH:
+        return
+
+    # Compute exit queue epoch [Modified in Electra:EIP7251]
+    exit_queue_epoch = compute_exit_epoch_and_update_churn(state, validator.effective_balance)
+
+    # Set validator exit epoch and withdrawable epoch
+    validator.exit_epoch = exit_queue_epoch
+    validator.withdrawable_epoch = Epoch(validator.exit_epoch + MIN_VALIDATOR_WITHDRAWABILITY_DELAY)
+```
+
+#### New `switch_to_compounding_validator`
+
+```python
+def switch_to_compounding_validator(state: BeaconState, index: ValidatorIndex) -> None:
+    validator = state.validators[index]
+    if has_eth1_withdrawal_credential(validator):
+        validator.withdrawal_credentials[:1] = COMPOUNDING_WITHDRAWAL_PREFIX
```

Comment:
```suggestion
        validator.withdrawal_credentials = COMPOUNDING_WITHDRAWAL_PREFIX + validator.withdrawal_credentials[1:]
```

It  gives an error otherwise

Reply:
Alternatively:
```suggestion
        validator.withdrawal_credentials[0] = COMPOUNDING_WITHDRAWAL_PREFIX
```

Reply:
Still doesn't work:`TypeError: 'SpecialByteVectorView' object does not support item assignment`

Reply:
going to copy the bls-to-exec creds processing fn from `capella`